src/Microsoft.Health.Fhir.Azure/ContainerRegistry/AzureContainerRegistryAccessTokenProvider.cs

```diff
@@ -0,0 +1,183 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Fhir.Core.Configs;
+using Microsoft.Health.Fhir.Core.Features.Operations.ConvertData;
+using Microsoft.Health.Fhir.Core.Features.Operations.ConvertData.Models;
+using Microsoft.Health.Fhir.Core.Features.Operations.Export.ExportDestinationClient;
+using Newtonsoft.Json;
+using Polly;
+
+namespace Microsoft.Health.Fhir.Azure.ContainerRegistry
+{
+    /// <summary>
+    /// Retrieve ACR accessTOkens with AAD token provider.
+    /// We need to exchange ACR refresh token with AAD token, and get ACR access token from refresh token.
+    /// References:
+    /// https://github.com/Azure/acr/blob/main/docs/AAD-OAuth.md#calling-post-oauth2exchange-to-get-an-acr-refresh-token
+    /// https://github.com/Azure/acr/blob/main/docs/AAD-OAuth.md#calling-post-oauth2token-to-get-an-acr-access-token
+    /// </summary>
+    public class AzureContainerRegistryAccessTokenProvider : IContainerRegistryTokenProvider
+    {
+        private const string ARMResourceUrl = "https://management.azure.com/";
```

Comment:
>ARMResourceUrl [](start = 29, length = 14)

I believe non public clouds (mooncake or forest) have different url, so maybe it's worth to push it to config?

Reply:
Have synced with ACR team: there are three possible Clouds: azure(includes ppe), mooncake and gov. I have added a new configuration for the cloud (by default Azure, can be configured to different clouds if needed)